### PR TITLE
Fix Travis CI

### PR DESCRIPTION
**Pull request by jaredsburrows**
@bamboo Can you please see: https://stackoverflow.com/a/31882307/950427 and then go here: https://travis-ci.org/gradle/kotlin-dsl and make sure that the build is configured correctly?
- Fix TravisCI
- Add badge to README.md

### CODE REVIEW

**Comment by jaredsburrows** [comment]
@bamboo After to toggle that switch, we should be able to trigger this build to run properly.

**Comment by jaredsburrows** [comment]
@bamboo I got it down to just a single test failure: https://travis-ci.org/gradle/kotlin-dsl/builds/258276536.

**Comment by eskatos** [comment]
Nice.

The failed test is: `org.gradle.kotlin.dsl.integration.KotlinBuildScriptModelIntegrationTest > sourcePath includes kotlin-stdlib sources resolved against project`

Cause is: `DaemonDisappearedException: Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)`

See: https://travis-ci.org/gradle/kotlin-dsl/builds/258276536#L2228-L2232

We have often seen Travis systems killing Gradle daemons during the integration tests when running out of resources. @jaredsburrows I couldn't find how, could you trigger another Travis build on the very same commit just to see if it happens always to the same test?

**Comment by jaredsburrows** [comment]
@eskatos Let me investigate. I have compiled projects that use a lot more memory on travis.

**Comment by JLLeitschuh** [comment]
Thank you figuring this out!!!!!

What do each of these new flags mean/do? How did you figure this out?!

**Comment by JLLeitschuh** [comment]
Something I'd love to see is this also testing GSK against Java 9.
Not something necessary in this PR. Could totally be a follow on.

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
+jdk:
+  - oraclejdk8
 
-script: ./gradlew :provider:check --info --stacktrace --console=plain --max-workers=1 --no-daemon -Dkotlin.compiler.execution.strategy=in-proces -Dkotlin.colors.enabled=false
+install: true
+
+script:
+  - ./gradlew clean customInstallation :provider:check --info --stacktrace --console=plain --max-workers=1 --no-daemon -Dkotlin.compiler.execution.strategy=in-proces -Dkotlin.colors.enabled=false
+
+sudo: false
```
Comment (JLLeitschuh) [inline .travis.yml]: I usually like to see this as the top line in the travis file.

**Comment by jaredsburrows** [comment]
@JLLeitschuh Do you have an example TravisCI configuration utilizing Java 9?

**Comment by jaredsburrows** [comment]
@bamboo @eskatos Can you make sure this is building on the CI? There is still not Github status. Thanks!

**Comment by JLLeitschuh** [comment]
It doesn't look like it is? Perhaps its because your PR isn't mergeable cleanly with master because its not pointed at the master branch?
I think travis won't run builds if the code won't merge with master without merge conflicts. But I could be wrong.
This is the PR list that has run. Nothing has run in the past 6 days it seems.
https://travis-ci.org/gradle/kotlin-dsl/pull_requests

[Travis: My Pull Request isn’t being built?](https://docs.travis-ci.com/user/pull-requests/#My-Pull-Request-isn%E2%80%99t-being-built)

Example of Java 9 cofig:
https://github.com/junit-team/junit5/blob/master/.travis.yml#L13-L16

I don't think that Java 9 will work given my experience with a Junit 5 PR I opened earlier this week:
https://travis-ci.org/junit-team/junit5/jobs/262456240#L680-L924
I'm 60% sure this is because the compiler that the kotlin DSL is using isn't compatible with Java 9.

**Comment by jaredsburrows** [comment]
They asked me to make PRs to the dev branch. I believe it is in the
Readme.md.

**Comment by JLLeitschuh** [comment]
Try pushing another simple whitespace change against the branch and see if it kicks the build. Or even modify your commit message and force push to this branch. Any change should re-trigger the build hooks.
It might not work though because I think travis uses the `.travis.yml` file on the target branch. Not 100% sure though. Give it a shot.

**Comment by jaredsburrows** [comment]
I have tried this with a new commit. I will try again soon.

**Comment by eskatos** [comment]
@jaredsburrows, @JLLeitschuh, thanks for looking into this!

I took some time to implement a solution I had in mind for quite a time. Turns out that was it. The Travis VMs were randomly killing processes when running out of resources. Our integration tests do spawn daemons, test workers and Kotlin compilation daemons. All this using default heap size values that most of the time were too big. Lowering these to sensible values allowed the build to pass on Travis infra.

The Travis build is now passing on the `develop` branch. I'm closing this PR.

See this series of commits;
- a88fd86aa767e2857002a33609e15efba09adeeb
- 39b14697a43b4d2908fc08a0273d0e355354da13
- 9bae8c4567becd887b398b59334f573d543efac6
- c3a2052bf27a56e7f9a2d47801602e461177da63
- e9ccf2c9916e0e4958c5bb25ec5c0df446b67f9c

The Travis badge wad also added to the README in 1217bf297356ccd3881260d4be7bb88f4d81927c

**Comment by jaredsburrows** [comment]
Its passing? No merge needed? When will you update the master?

**Comment by eskatos** [comment]
@jaredsburrows I pushed the necessary changes to the `develop` branch where work happens.
It will be merged to `master` on the next release.

**Comment by jaredsburrows** [comment]
@eskatos Thank you so much for your updates!

**Pull request closed without merging**